scalameta/common/shared/src/main/scala/scala/meta/internal/trees/ast.scala

```diff
@@ -720,4 +735,25 @@ object AstNamerMacros {
     if (majorIdx > 0) Some(buildVersion.substring(0, majorIdx).toInt) else None
   }
 
+  private val initialName = "Initial"
```

Comment:
what's the need for that extractor? isn't `case Foo.Initial(xxx)` the same as `case Foo(xxx)`?

Reply:
it is today but in 5.0.0 it won't be. in fact, I'd like to discourage `case Foo(xxx)` and might suggest completely removing the non-versioned `unapply` completely in next major.

Reply:
another option is to remove Initial, remove deprecation from original Term.Apply but then remove the unapply method from it in the next major (if there are any intervening versions). would that work better?